tests/live/test_userid.py

```diff
@@ -56,28 +109,37 @@ def test_08_get_registered_ip(self, fw, state_map):
         if not state.multi_register_01:
             raise Exception("Multi register not set")
         ips, tags = state.multi_register_01
-        test1 = set(fw.userid.get_registered_ip())
-        assert test1 == set(ips)
-        test2 = set(fw.userid.get_registered_ip(
+        test1 = fw.userid.get_registered_ip()
+        assert set(test1) == set(ips)
+        for x in ips:
+            assert set(test1[x]) == set(tags)
+        test2 = fw.userid.get_registered_ip(
                     ips[0:3], tags
-        ))
-        assert test2 == set(ips[0:3])
-        test3 = set(fw.userid.get_registered_ip(
+        )
+        assert set(test2) == set(ips[0:3])
+        for i in ips[0:3]:
+            assert set(test2[i]) == set(tags)
+        test3 = fw.userid.get_registered_ip(
                     ips[0:3], tags[0:5]
-        ))
-        assert test3 == set(ips[0:3])
-        test4 = set(fw.userid.get_registered_ip(
+        )
+        assert set(test3) == set(ips[0:3])
+        for i in ips[0:3]:
+            assert set(test3[i]) == set(tags[0:5])
+        test4 = fw.userid.get_registered_ip(
                     ips, tags[0:5]
-        ))
-        assert test4 == set(ips)
-        test5 = set(fw.userid.get_registered_ip(ips[0], tags[0]))
-        assert test5 == set([ips[0], ])
+        )
+        assert set(test4) == set(ips)
+        for x in ips:
+            assert set(test4[x]) == set(tags[0:5])
+        test5 = fw.userid.get_registered_ip(ips[0], tags[0])
+        assert list(test5.keys()) == [ips[0], ]
+        assert test5[ips[0]] == [tags[0], ]
         tests = [test1, test2, test3, test4, test5]
         assert len(test5) != 0
-        assert all([test1 >= x for x in tests])
-        assert all([x >= test5 for x in tests])
-        assert test2 >= test3
-        assert test4 >= test3
+        assert all([set(test1) >= set(x) for x in tests])
+        assert all([set(x) >= set(test5) for x in tests])
+        assert set(test2) >= set(test3)
+        assert set(test4) >= set(test3)
```

Comment:
I would personally never compare sets like this, but instead use `.issuperset()` / `.issubset()`.

But looking over just how big test7 is, I think this needs to be split into multiple tests, each test named by what type of `get_registered_ip()` type it's doing:  single, multiple, etc.